### ISSUE TITLE
Add support for Sets

### ISSUE THREAD

**Issue by ethanblake4** [ethanblake4/dart_eval#2]
This will need:
Support in the parser (parser.dart)
A new EvalSet type in primitives.dart
A new DartSetLiteral class in literals.dart
???

**Comment by wrbl606** [comment]
Hi, are the points above still applicable? I'm planning to contribute and think that sets would be a good start.

**Comment by ethanblake4** [comment]
Ah not really, this issue is quite old.
For Sets, you'd probably need to:
- edit [collection/set_map.dart](https://github.com/ethanblake4/dart_eval/blob/master/lib/src/eval/compiler/collection/set_map.dart) in the compiler,
- add a $Set wrapper class in `shared` similar to the [$Map](https://github.com/ethanblake4/dart_eval/blob/master/lib/src/eval/shared/stdlib/core/map.dart) and [$List](https://github.com/ethanblake4/dart_eval/blob/master/lib/src/eval/shared/stdlib/core/list.dart) wrappers, 
- add `PushSet`, `SetSet`, and `IndexSet` bytecode ops similar to [the Map equivalents](https://github.com/ethanblake4/dart_eval/blob/e533b782083b35d2cf7390b4010a39a28884bf33/lib/src/eval/runtime/ops/primitives.dart#L414) in the runtime,
- and generate code to get/set values from the Set in [IndexedReference](https://github.com/ethanblake4/dart_eval/blob/e533b782083b35d2cf7390b4010a39a28884bf33/lib/src/eval/compiler/reference.dart#L345).

To be frank this is slightly complicated for a first issue at this point since it touches all three parts of dart_eval (compiler, standard library, and runtime). It is doable but I might suggest #4 or #91 first. (Thanks for your interest in contributing btw!)

**Comment by wrbl606** [comment]
You're right, I thought that it'll be easier to do (somehow) 😀 Thanks for the explanation!

**Comment by Noobware1** [comment]
Hi, @ethanblake4 i can do the 2 and 3 points you said.

The 1st and 4th are just beyond me. If you could elaborate just on them 🙏🙏 it would be big help.
